lib/plugins/digging.js

```diff
@@ -110,20 +121,30 @@ function inject (bot) {
       location: block.position,
       face: diggingFace // default face is 1 (top)
     })
-    const waitTime = bot.digTime(block)
-    waitTimeout = setTimeout(finishDigging, waitTime)
+    if (waitInterval) { clearInterval(waitInterval) }
+    waitInterval = setInterval(() => {
+      bot.digPercentage += (50 / bot.digTime(block))
```

Comment:
does the vanilla client follow this algorithm when state change while digging ? do you have a reference ?

Reply:
Probably not this exact algorithm but it mimics what I've seen in game, and I've yet to see something abnormal from it.